occurrence-download/src/main/java/org/gbif/occurrence/download/file/specieslist/SpeciesListCollector.java

```diff
@@ -26,25 +28,20 @@
   
   /**
    * group results by taxon key and order them in {@link DownloadTerms} species list download order.
-   * @return distinct species
    */
-  public void computeDistinctSpecies(Map<String, String> filteredResult) {
-    String taxonKey = filteredResult.get(GbifTerm.taxonKey.simpleName());
+  public void collect(Map<String, String> occurrenceRecord) {
+    String taxonKey = occurrenceRecord.get(GbifTerm.taxonKey.simpleName());
 
     distinctSpeciesRecord.put(taxonKey, distinctSpeciesRecord.compute(taxonKey, (k, v) -> {
       //if the values are already there increment
       if (v != null) {
         long count = Long.parseLong(v.get(GbifTerm.numOfOccurrences.simpleName())) + 1L;
         v.put(GbifTerm.numOfOccurrences.simpleName(), Long.toString(count));
         return v;
-      } 
-      else {
-        Map<String, String> orderedResults = new LinkedHashMap<>();
-        filteredResult.put(GbifTerm.numOfOccurrences.simpleName(), Long.toString(1L));
+      } else {
+        occurrenceRecord.put(GbifTerm.numOfOccurrences.simpleName(), Long.toString(1L));
```

Comment:
it is not a good practice to modify params, in this case we have control of the code but one can not rely that the map is modifiable, better assume that this object is immutable